apps/jobs/views.py

```diff
@@ -745,3 +745,35 @@ def update_submission(request, challenge_pk):
     submission.save()
     response_data = {'success': 'Submission result has been successfully updated'}
     return Response(response_data, status=status.HTTP_200_OK)
+
+
+@api_view(['GET', ])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_submissions_for_challenge(request, challenge_pk):
+
+    challenge = get_challenge_model(challenge_pk)
+
+    if not is_user_a_host_of_challenge(request.user, challenge.id):
+        response_data = {'error': 'Sorry, you are not authorized to make this request!'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    submission_status = request.query_params.get('status', None)
+
+    valid_submission_status = [Submission.SUBMITTED, Submission.RUNNING, Submission.FAILED,
```

Comment:
So this status check should be a part of the serializer and not in the view. Serializers are meant to do the data validation. Can you please move this in the serializer?

Reply:
My bad. This is not a valid comment. Please use django-filter instead of manually validating the status.